charm_test.go

```diff
@@ -219,3 +225,88 @@ func cloneDir(c *gc.C, path string) string {
 	c.Assert(err, gc.IsNil)
 	return newPath
 }
+
+func (s *CharmSuite) assertVersionFile(c *gc.C, dir string, execName string, args []string) {
+	testing.PatchExecutableAsEchoArgs(c, s, execName)
+
+	// copy all the contents from 'path' to 'tmp folder dummy-charm'
+	// Using cloneDir
+	tempPath := cloneDir(c, dir)
+
+	// create an empty .execName file inside tempDir
+	lookupDir := "." + execName
+	vcsPath := filepath.Join(tempPath, lookupDir)
+	_, err := os.Create(vcsPath)
+	c.Assert(err, gc.IsNil)
+
+	err = charm.MaybeCreateVersionFile(tempPath)
+	c.Assert(err, gc.IsNil)
+
+	testing.AssertEchoArgs(c, execName, args...)
+
+	// Verify if version exists.
+	versionPath := filepath.Join(tempPath, "version")
+	_, err = os.Stat(versionPath)
+	c.Assert(err, gc.IsNil)
+
+	// Read the contents of version file.
+	var version []string
```

Comment:
Don't declare this all the way up here when it's not used till down below. Try and declare variables close to where they're used